plugins/Files/js/components/renamedialog.js

```diff
@@ -0,0 +1,34 @@
+import React, { PropTypes } from 'react'
+
+const RenameDialog = ({siapath, actions}) => {
+	const onYesClick = (e) => {
+		e.preventDefault()
+		actions.renameFile(siapath, e.target.newsiapath.value)
+		actions.hideRenameDialog()
+	}
+	const onNoClick = () => actions.hideRenameDialog()
+	return (
+		<div className="modal">
+			<div className="rename-dialog">
+				<div className="rename-text">
+					Enter a new name for {siapath}:
+				</div>
+				<form className="rename-form" onSubmit={onYesClick}>
+					<div className="rename-field">
+						<input type="text" name="newsiapath" required autoFocus defaultValue={siapath} />
```

Comment:
So this rename dialog automatically fills with the current full siapath of the file you're renaming.  This could be confusing if the users select all, then type a different name, and the file moves out of the folder it was in.  I think `RenameDialog` should only rename the _base name_ of the file, you can use `Path.basename` and `Path.dirname` to accomplish this, in the `onYesClick` function.